### ISSUE TITLE
Filter a DateOnly column without the time

### ISSUE THREAD

**Issue by Aymkdn** [pnp/pnpjs#1890]
### Category
- [ ] Enhancement
- [ ] Bug
- [x] Question
- [ ] Documentation gap/issue

### Version

PNPJS v2.9.0

Target: Sharepoint Online

### Expected / Desired Behavior / Question

I have several date-only columns (no time can be selected) that return either "**YYYY-MM-DDT05:00:00Z**" or "**YYYY-MM-DDT06:00:00Z**", which makes things odd and difficult to deal with.

Example: 
![image](https://user-images.githubusercontent.com/946315/135464014-c402ee09-3648-400b-89ea-cea4aba0d180.png)

I don't know why the timepart is different…

How can I filter these columns with only using the date part, and without worrying about the time part? With SOAP Web Services we can just provide the date part.

For example, if my field shows: "2021-09-19" I'd like to do the below filter:
>  .filter("ColName eq datetime'2021-09-19'")

Using a date range to cover the whole day (.e.g `ColName ge datetime'2021-09-19T00:00:00Z'` and `ColName lt datetime'2021-09-20T00:00:00Z'`) won't probably work with a large list that could return more than 5,000 items.

My current solution is to test both `T05:00:00Z` and `T06:00:00Z` but it looks like a dirty hack, and I'm not sure this is tied to my own timezone?!

Thanks

**Comment by Aymkdn** [comment]
So there is no way to filter by a date only?

Do you see the same behavior in your Sharepoint? Is it universal? Meaning that all Sharepoint Online sites, whatever how the settings are defined, will have the date-only columns with the time defined on both `T05:00:00Z` and `T06:00:00Z`?

**Comment by koltyakov** [comment]
Yes, it's a default behaviour. Zero time for web's timezone stored in UTC. Don't forget daylight time saving aspects.

Not sure what's the best option here. Yet, date-range filter should be throttled on a large list in most of the cases.

Also, as you know web's time zone and needs filtering by a day in the past, taking web's 0hh time and getting UTC should ends up with the time stored in SP.

I guess ignoring time part is only possible with a CAML query.
